sdk/src/main/java/com/meniga/sdk/models/sync/MenigaSync.kt

```diff
@@ -0,0 +1,238 @@
+                override fun onFinished(result: MenigaSyncStatus?, failed: Boolean) {
+
+                    val delta = DateTime.now().millis - postSync.startTimeStamp!!
+                    if (failed || result == null) {
+                        postSync.onFailure(Exception("Failure syncing"))
+                    } else if (result.getHasCompletedSyncSession()) {
+                        val subTask = MenigaSync.fetch(syncHistoryId)
+                        MenigaSDK.getMenigaSettings().taskAdapter.intercept(subTask, object : Interceptor<MenigaSync>() {
+                            override fun onFinished(result: MenigaSync?, failed: Boolean) {
+                                if (failed || result == null) {
+                                    postSync.onFailure(Exception("Failure syncing"))
+                                }
+                                postSync.hasNewData = result != null && result.hasNewData()
+                                postSync.numNewTransactions = result?.numNewTransactions ?: 0
+                                postSync.sync = result
+                                postSync.onFinished()
+                            }
+                        })
+                    } else if (delta > timeout) {
+                        postSync.onTimedOut()
+                    } else {
+                        checkSync(postSync, timeout, interval)
+                    }
+                }
+            })
+        }, interval)
+    }
+
+    private fun hasNewData(): Boolean = realmSyncResponses
+            ?.flatMap { it.getAccountSyncStatuses() }
+            ?.firstOrNull { it.getTransactionsProcessed() > 0 } != null
+
+    /**
+     * Checks to see if the synchronization operation has finished
+     *
+     * @return A Task containing a boolean indicating if the synchronization is done or not
+     */
+    @Deprecated("Reason this method is deprecated </br>\n" +
+            "\t               {Will be removed in next version} </br>\n" +
+            "\t               use static {@link #getSyncStatus()} instead like this:\n" +
+            "\t \n" +
+            "\t  <blockquote>\n" +
+            "\t  <pre>\n" +
+            "\t  MenigaSync.getSyncStatus()\n" +
+            "\t  </pre></blockquote>")
+    fun isSyncDone(): Result<MenigaSyncStatus> = apiOperator.syncStatus
+
+    companion object {
+        private const val SYNC_CHECK_INTERVAL_MILLISECONDS: Long = 1000
+
+        private lateinit var apiOperator: MenigaSyncOperations
+
+        /**
+         * Sets the api operator for doing api calls
+         *
+         * @param operator An object that implements the MenigaSyncOperations interface for carrying out api operations on this class.
+         */
+        @JvmStatic
+        fun setOperator(operator: MenigaSyncOperations) {
+            MenigaSync.apiOperator = operator
+        }
+
+        /**
+         * Gets a sync object that was created by MenigaSync.start
```

Comment:
Update comment to point to MenigaSync.syncRealms

Reply:
Done